Scenario for Azure Arc-enabled Kubernetes - Connected Cluster with an existing cluster (#1498)
microsoft/azure_arc: docs/azure_arc_jumpstart/azure_arc_k8s/day2/multi_distributions/connected_cluster/_index.md

---
type: docs
title: "Using Cluster Connect to connect to an Azure Arc-enabled Kubernetes cluster"
linkTitle: "Using Cluster Connect to connect to an Azure Arc-enabled Kubernetes cluster"

> weight: 4

Reviewer: ```suggestion
weight: 5
```

description: >
---

## Using Cluster Connect to connect to an Azure Arc-enabled Kubernetes cluster

> The following Jumpstart scenario will guide you through how to use the [cluster connect](https://docs.microsoft.com/azure/azure-arc/kubernetes/cluster-connect) capability of an Azure Arc connected Kubernetes cluster to manage and connect to the cluster securely without opening inbound firewall ports.

Reviewer: ```suggestion
The following Jumpstart scenario will guide you through how to use the [Cluster Connect](https://docs.microsoft.com/azure/azure-arc/kubernetes/cluster-connect) capability of an Azure Arc connected Kubernetes cluster to manage and connect to the cluster securely without opening inbound firewall ports.
```

- **[AKS on Azure Stack HCI](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/aks_stack_hci/)**
- **[Kubernetes Cluster API](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/cluster_api/)**
- **[Azure Red Hat OpenShift](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/aro/)**
- **[Amazon Elastic Kubernetes Service](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/eks/)**
- **[Google Kubernetes Engine](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/gke/)**
- **[Alibaba Cloud Container Service for Kubernetes](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/alibaba/)**
- **[Rancher K3s](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/rancher_k3s/)**
- **[Kind](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/kind/)**
- **[MicroK8s](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/microk8s/)**
- **[Platform9 Managed Kubernetes](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/pf9/)**

## Prerequisites

- Clone the Azure Arc Jumpstart repository

    ```shell
    git clone https://github.com/microsoft/azure_arc.git
    ```

> - Ensure you are logged in to the Azure CLI 

Reviewer: The README assumes shell but what about PowerShell?

Author: I added PowerShell commands where applicable

  ```shell
  az login
  ```

> - [Install or update Azure CLI to version 2.36.0 and above](https://docs.microsoft.com/cli/azure/install-azure-cli?view=azure-cli-latest). Use the below command to check your current installed version.

Reviewer: ```suggestion
- [Install or update Azure CLI to version 2.42.0 and above](https://docs.microsoft.com/cli/azure/install-azure-cli?view=azure-cli-latest). Use the below command to check your current installed version.
```

  ```shell
  az --version
  ```

> - [Install or update to the latest version of the *connectedk8s* CLI extension](https://learn.microsoft.com/azure/azure-arc/kubernetes/cluster-connect?tabs=azure-cli).

Reviewer: ```suggestion
- [Install or update to the latest version of the _connectedk8s_ CLI extension](https://learn.microsoft.com/azure/azure-arc/kubernetes/cluster-connect?tabs=azure-cli).
```

  ```shell
  az extension add --name connectedk8s
  ```

- As mentioned, this scenario starts at the point where you already have an Azure Arc connected cluster.

>     ![Existing Azure Arc connected cluster](./01.png)

Reviewer: ```suggestion
    ![Screenshot showing existing Azure Arc connected cluster](./01.png)
```

## Azure Arc Kubernetes Connected Cluster Configuration

> The following steps walk through using the connected cluster functionality using Azure Active Directory; however, it can also be done using a service account token.  The benefit of using Azure Active Directory authentication is that your current, logged in user from the Azure CLI will be used and you won't have to obtain a service account token from your cluster.

Reviewer: ```suggestion
The following steps walk through using the Cluster Connect functionality using Azure Active Directory (AAD); however, it can also be done using a service account token.  The benefit of using AAD authentication is that your current, logged-in user from the Azure CLI will be used and you won't have to obtain a service account token from your cluster.
```

> - First, get values for the cluster name and resource group name in your terminal

Reviewer: ```suggestion
- First, get values for the cluster name and resource group name in your terminal.
```

  ```shell
  CLUSTER_NAME=<cluster-name>
  RESOURCE_GROUP=<resource-group-name>
  ```

> - Second, obtain the *user principal name and ID* of your Azure Active Directory user account

Reviewer: ```suggestion
- Second, obtain the _user principal name and ID_ of your AAD user account.
```

  ```shell
  AAD_ENTITY=$(az ad signed-in-user show --query "[id, userPrincipalName]" -o tsv)
  ```

> - Finally, create the ClusterRoleBinding using the following:

Reviewer: ```suggestion
- Finally, create the Kubernetes _ClusterRoleBinding_ using the following:
```

  ```shell
  while ID= read -r line; do  kubectl create clusterrolebinding jumpstart-binding-$line --clusterrole cluster-admin --user=$line; done <<< "$AAD_ENTITY"
  ```

## Access Cluster

From your terminal, run the following command to establish the proxy to the cluster.  Note that the specific port number that's used may differ from the screenshot below.

  ```shell
  az connectedk8s proxy -n $CLUSTER_NAME -g $RESOURCE_GROUP
  ```

>   ![Proxy established](./02.png)

Reviewer: ```suggestion
  ![Screenshot showing existing Proxy established](./02.png)
```

> ## Run Kubectl Commands

Reviewer: ```suggestion
## Run kubectl commands
```

> With the proxy established, in another terminal session run a kubectl.

Reviewer: ```suggestion
With the proxy established, in another terminal session run a kubectl command.
```

  ```shell
  kubectl get nodes
  ```

>   ![Running kubectl command](./03.png)

Reviewer: ```suggestion
  ![Screenshot showing existing a running kubectl command](./03.png)
```

> If you already have a workload running on your cluster, you can view it using other kubectl commands, like the following screenshot illustrates:

Reviewer: ```suggestion
If you already have a workload running on your cluster, you can view it using other kubectl commands, as the following screenshot illustrates:
```

>   ![Example kubectl command](./04.png)

Reviewer: ```suggestion
  ![Screenshot showing existing an example kubectl command](./04.png)
```

## Close Proxy Connection

> In your terminal, press *CTRL+C* to close the proxy session.

Reviewer: ```suggestion
In your terminal, use the _CTRL+C_ key combination to close the proxy session.
```

>   ![Closing Proxy to connected cluster](./05.png)

Reviewer: ```suggestion
  ![Screenshot showing closing Proxy to the connected cluster](./05.png)
```

- **[AKS on Azure Stack HCI](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/aks_stack_hci/)**
- **[Kubernetes Cluster API](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/cluster_api/)**
- **[Azure Red Hat OpenShift](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/aro/)**
- **[Amazon Elastic Kubernetes Service](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/eks/)**
- **[Google Kubernetes Engine](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/gke/)**
- **[Alibaba Cloud Container Service for Kubernetes](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/alibaba/)**
- **[Rancher K3s](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/rancher_k3s/)**
- **[Kind](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/kind/)**
- **[MicroK8s](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/microk8s/)**
- **[Platform9 Managed Kubernetes](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/pf9/)**

## Prerequisites

- Clone the Azure Arc Jumpstart repository

    ```shell
    git clone https://github.com/microsoft/azure_arc.git

>     ```

Reviewer: Do we need this in this scenario?

Author: git clone has been removed

  az login
  ```

  ```powershell
  Connect-AzAccount
  ```

- [Install or update Azure CLI to version 2.42.0 and above](https://docs.microsoft.com/cli/azure/install-azure-cli?view=azure-cli-latest). Use the below command to check your current installed version.

  ```shell
  az --version
  ```

- [Install or update to the latest version of the _connectedk8s_ CLI extension](https://learn.microsoft.com/azure/azure-arc/kubernetes/cluster-connect?tabs=azure-cli).

  ```shell
  az extension add --name connectedk8s
  ```

- As mentioned, this scenario starts at the point where you already have an Azure Arc connected cluster.

    ![Screenshot showing existing Azure Arc connected cluster](./01.png)

## Azure Arc Kubernetes Connected Cluster Configuration

The following steps walk through using the Cluster Connect functionality using Azure Active Directory (AAD); however, it can also be done using a service account token.  The benefit of using AAD authentication is that your current, logged-in user from the Azure CLI will be used and you won't have to obtain a service account token from your cluster.

- First, get values for the cluster name and resource group name in your terminal.

>   ```shell
>   CLUSTER_NAME=<cluster-name>
>   RESOURCE_GROUP=<resource-group-name>
>   ```
> 
>   ```powershell
>   $CLUSTER_NAME = <cluster name>
>   $RESOURCE_GROUP = <resource_group>
>   ```

Reviewer: ```suggestion
Bash:

  ```shell
  CLUSTER_NAME=<cluster-name>
  RESOURCE_GROUP=<resource-group-name>
  ```

PowerShell:
  ```powershell
  $CLUSTER_NAME = <cluster name>
  $RESOURCE_GROUP = <resource_group>
  ```
```

> 
>   ```powershell
>   $CLUSTER_NAME = <cluster name>
>   $RESOURCE_GROUP = <resource_group>
>   ```
> 

Reviewer: ```suggestion

PowerShell:
  ```powershell
  $CLUSTER_NAME = <cluster name>
  $RESOURCE_GROUP = <resource_group>
  ```

```

Reviewer: Wherever applicable, add Bash/PowerShell command indicator before the command. Otherwise, one can be confused by the two code blocks

  ```shell
  az --version
  ```

- [Install or update to the latest version of the _connectedk8s_ CLI extension](https://learn.microsoft.com/azure/azure-arc/kubernetes/cluster-connect?tabs=azure-cli).

  ```shell
  az extension add --name connectedk8s
  ```

- As mentioned, this scenario starts at the point where you already have an Azure Arc connected cluster.

    ![Screenshot showing existing Azure Arc connected cluster](./01.png)

## Azure Arc Kubernetes Connected Cluster Configuration

The following steps walk through using the Cluster Connect functionality using Azure Active Directory (AAD); however, it can also be done using a service account token.  The benefit of using AAD authentication is that your current, logged-in user from the Azure CLI will be used and you won't have to obtain a service account token from your cluster.

- First, get values for the cluster name and resource group name in your terminal.

  #### Option 1: Bash

  ```shell
  CLUSTER_NAME=<cluster-name>
  RESOURCE_GROUP=<resource-group-name>
  ```

  #### Option 2: PowerShell

>   ```powershell
>   $CLUSTER_NAME = <cluster name>
>   $RESOURCE_GROUP = <resource_group>

Reviewer: ```suggestion
  ```powershell
  $CLUSTER_NAME = <cluster name>
  $RESOURCE_GROUP = <resource_group>
  ```
```

- Second, obtain the _user principal name and ID_ of your AAD user account.

  #### Option 1: Bash

  ```shell
  AAD_ENTITY=$(az ad signed-in-user show --query "[id, userPrincipalName]" -o tsv)
  ```

  #### Option 2: PowerShell

  ```powershell
  $AAD_ENTITY = (az ad signed-in-user show --query "[id, userPrincipalName]" -o tsv)
  ```

- Finally, create the Kubernetes _ClusterRoleBinding_ using the following:

  #### Option 1: Bash

  ```shell
  while ID= read -r line; do  kubectl create clusterrolebinding jumpstart-binding-$line --clusterrole cluster-admin --user=$line; done <<< "$AAD_ENTITY"
  ```

  #### Option 2: PowerShell

  ```powershell
  foreach($line in $AAD_ENTITY)
  { kubectl create clusterrolebinding jumpstart-binding-$line --clusterrole cluster-admin --user=$line }
  ```

## Access Cluster

From your terminal, run the following command to establish the proxy to the cluster.  Note that the specific port number that's used may differ from the screenshot below.

  #### Option 1: Bash

  ```shell
  az connectedk8s proxy -n $CLUSTER_NAME -g $RESOURCE_GROUP
  ```

  #### Option 2: PowerShell
  
  ```powershell
  az connectedk8s proxy -n $CLUSTER_NAME -g $RESOURCE_GROUP

>   ```

Author: It's unneeded and has been removed
